File: 백준/Silver/11399. ATM/ATM.py
import sys
def input():return sys.stdin.readline().rstrip()
MAX=sys.maxsize
MOD=int(1e9)+9
N=int(input())
a=list(map(int,input().split()))
a.sort()
res=0
c=0
for cur in a:
    c+=cur
    res+=c
print(res)

File: 백준/Silver/11399. ATM/ATM.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(br.readLine());
        int[] a = new int[N];
        
        String[] input = br.readLine().split(" ");
        for (int i = 0; i < N; i++) {
            a[i] = Integer.parseInt(input[i]);
        }
        
        Arrays.sort(a);
        
        long res = 0;
        long c = 0;
        for (int cur : a) {
            c += cur;
            res += c;
        }
        
        System.out.println(res);
    }
}
